lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,344 @@
+    return {
+        "data": batch_item_failures,
+        "statusCode": 400 if batch_item_failures else 200,
+    }
+
+
+def create_response(status_code: int):
+    """
+    Create a response object to return after the lambda completes processing.
+    """
+
+    return {
+        "statusCode": status_code,
+        "isBase64Encoded": False,
+        "body": "Success" if status_code == 200 else "Failure"
+    }
+
+
+def valid_message_body(event_data: dict) -> bool:
+    """
+    Return a boolean to indicate if a record body's message contains
+    all required attributes.
+    """
+
+    return EXPECTED_PINPOINT_FIELDS.issubset(event_data)
+
+
+def valid_event(event_data: dict) -> bool:
+    """
+    Ensure the event data has a "Records" list and that each record has a
+    "body" attribute.
+    """
+
+    if event_data is None or "Records" not in event_data:
+        return False
+
+    return all((record.get("body") is not None) for record in event_data["Records"])
+
+
+def forward_to_service(inbound_sms: dict, url: str) -> bool:
+    """
+    Forwards the inbound SMS to the service that has 2-way SMS setup.
+    """
+
+    # This covers None and the empty string.
+    if not url:
+        logger.error("No URL provided in the configuration for the service.")
+        return False
+
+    headers = {
+        'Content-type': 'application/json'
+    }
+
+    try:
+        logger.debug('Connecting to %s, sending: %s', url, inbound_sms)
+        response = requests.post(
+            url,            
+            verify=False,
+            json=inbound_sms,
+            timeout=TIMEOUT,
+            headers=headers
+        )
+        logger.info('POST to service complete')
+        response.raise_for_status()
+
+        logger.info('Response Status: %d', response.status_code)
+        logger.debug('Response Content: %s', response.content)
+        return True
+    except requests.HTTPError as e:
+        logger.error("HTTPError With Http Request")
+        logger.exception(e)
+    except requests.RequestException as e:
+        logger.error("RequestException With Http Request")
+        logger.exception(e)
+    except Exception as e:
+        logger.error("General Exception With Http Request")
+        logger.exception(e)
+        # Need to raise here to move the message to the dead letter queue instead of retry. 
+        raise
+
+    return False
+
+
+def push_to_sqs(inbound_sms: dict, is_retry: bool) -> None:
+    """
+    Pushes an inbound sms or entire event to SQS. Sends to RETRY or DEAD LETTER queue dependent
+    on is_retry variable. 
+    """
+
+    # NOTE: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+    # Need to ensure none of those unicode characters are in the message or it's gone.
+    try:
+        logger.warning('Pushing event to %s queue', "RETRY" if is_retry else "DEAD LETTER")
+        logger.debug('Event: %s', inbound_sms)
+
+        queue_msg = json.dumps(inbound_sms)
```

Comment:
SNS messages are encoded. I believe we should be encoding before we send as well?

Reply:
I misread the SNS docs. SNS signatures are base64 encoded, and the signature includes the message field. We should not need to encode/decode the message.